src/api/PluginManager.ts

```diff
@@ -43,28 +63,28 @@ export async function startPlugins() {
                 zip.closeEntry();
 
                 if (manifest.name in plugins) throw new Error(`Plugin ${manifest.name} already registered`);
-                if (!isPluginEnabled(manifest.name)) continue;
+                if (!isPluginEnabled(manifest.name)) {
+                    disabledPlugins[manifest.name] = manifest;
+                    continue;
+                }
 
-                zip.openEntry("index.js.bundle");
-                const pluginBuffer = zip.readEntry("binary");
-                zip.closeEntry();
+                const pluginBuffer = loadPluginBundle(zip);
 
                 const pluginClass = AliuHermes.run(file.name, pluginBuffer) as typeof Plugin;
                 try {
                     if (pluginClass.prototype instanceof Plugin) {
                         if (manifest.name !== pluginClass.name) throw new Error(`Plugin ${manifest.name} must export a class named ${manifest.name}`);
                         logger.info(`Loading Plugin ${manifest.name}...`);
-                        const plugin = plugins[manifest] = new pluginClass(manifest);
+                        const plugin = plugins[manifest.name] = new pluginClass(manifest);
                         plugin.start();
```

Comment:
same here, except using await will make it so that plugins are loaded strictly sequentially so you should use .catch()